### PR TITLE
Calculation of compensation days

### PR DESCRIPTION
**Pull request by omdelafuente**
Added corresponding compensation days calculation depending on hiring date and fixed bug when calculating holidays for new users.

### CODE REVIEW

File: tntconcept-core/src/main/java/com/autentia/tnt/manager/holiday/UserHolidaysStateManager.java [hunk 1]
```diff
@@ -41,221 +45,228 @@
...
+		if (chargeYear != null) {
+
 			HolidayManager fiestasManager = HolidayManager.getDefault();
-			
+
 			// We must take in account previous year holidays and next year holidays
-			
-		Calendar calMin = Calendar.getInstance();
+
+			Calendar calMin = Calendar.getInstance();
```
Comment (angelusGJ) [inline tntconcept-core/src/main/java/com/autentia/tnt/manager/holiday/UserHolidaysStateManager.java]: Refactor this code and create a method to create the instance calendar with the year

File: tntconcept-core/src/test/java/com/autentia/tnt/manager/holiday/CorrespondingHolidayManagerTest.java [hunk 2]
```diff
@@ -0,0 +1,220 @@
...
+		LocalDate hiringDate = new GregorianCalendar(2018, Calendar.JANUARY, 10).getTime().toInstant().atZone(ZoneId.systemDefault()).toLocalDate();
+		
+		List<Holiday> result = sut.getCorrespondingHolidays(holidays, hiringDate, 2019);
+		
+		assertThat(result, is(not(empty())));
+	}
+	
+	@Test
+	public void shouldConsiderWeekendsAndHolidaysForCalculation() {
+		Holiday compensation1 = new Holiday();
```
Comment (angelusGJ) [inline tntconcept-core/src/test/java/com/autentia/tnt/manager/holiday/CorrespondingHolidayManagerTest.java]: Refactor this code to a method

File: tntconcept-core/src/main/java/com/autentia/tnt/manager/holiday/UserHolidaysStateManager.java (hunk 1 above)
Comment (omdelafuente) [reply tntconcept-core/src/main/java/com/autentia/tnt/manager/holiday/UserHolidaysStateManager.java]: Done

File: tntconcept-core/src/test/java/com/autentia/tnt/manager/holiday/CorrespondingHolidayManagerTest.java (hunk 2 above)
Comment (omdelafuente) [reply tntconcept-core/src/test/java/com/autentia/tnt/manager/holiday/CorrespondingHolidayManagerTest.java]: Done

**Pull request merged**